packages/fast-components-react-base/src/listbox/listbox.spec.tsx

```diff
@@ -0,0 +1,246 @@
+import * as React from "react";
+import * as Adapter from "enzyme-adapter-react-16";
+import { configure, mount, shallow } from "enzyme";
+import Listbox, { ListboxUnhandledProps } from "./listbox";
+import { KeyCodes } from "@microsoft/fast-web-utilities";
+
+/*
+ * Configure Enzyme
+ */
+configure({ adapter: new Adapter() });
+
+describe("listbox", (): void => {
+    test("should have a displayName that matches the component name", () => {
+        expect((Listbox as any).name).toBe(Listbox.displayName);
+    });
+
+    test("should not throw if managedClasses are not provided", () => {
+        expect(() => {
+            shallow(<Listbox />);
+        }).not.toThrow();
+    });
+
+    test("should implement unhandledProps", (): void => {
+        const unhandledProps: ListboxUnhandledProps = {
+            "aria-label": "label",
+        };
+
+        const rendered: any = shallow(<Listbox {...unhandledProps} />);
+
+        expect(rendered.first().prop("aria-label")).toEqual("label");
+    });
+
+    test("should not have a focusIndex if no children are focusable", (): void => {
+        const rendered: any = mount(
+            <Listbox>
+                <div>hello world</div>
+            </Listbox>
+        );
+
+        expect(rendered.state("focusIndex")).toBe(-1);
+    });
+
+    test("should set focusIndex to the first focusable element on mount", (): void => {
+        const rendered: any = mount(
+            <Listbox>
+                <div>not a focusable element</div>
+                <div role="option">focusable element</div>
+            </Listbox>
+        );
+
+        expect(rendered.state("focusIndex")).toBe(1);
+    });
+
+    test("should move focus down when the down arrow is pressed", (): void => {
```

Comment:
please add a test for a single child, a cursory view looks like because you are using the index you assume children are in an array, however a single react child is usually an object.

Reply:
Added and passed without code changes.  I think that because I always run children through one of the React node functions (map, filter or toArray) as a first action and I expect they handle the single child case appropriately.